xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-ui/src/main/resources/XWiki/Like/UserProfileUIX.xml

```diff
@@ -254,30 +254,53 @@
     </property>
     <property>
       <content>{{velocity}}
-=$escapetool.xml($services.localization.render("like.userprofile.menu"))=
-#set($columns = ["doc.title", "doc.location", "like"])
+=$escapetool.xml($services.localization.render('like.userprofile.menu'))=
 #set ($docTitleDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.title')))
 #set ($docLocationDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.location')))
 #set ($likesDisplayName = $escapetool.xml($services.localization.render('like.livetable.column.likes')))
-#set($columnsProperties = {
-  "doc.title" : { "displayName": $docTitleDisplayName, "type": "text", "size": 30, "link": "view", "sortable": false, "filterable": false },
-  "doc.location" : { "displayName": $docLocationDisplayName, "type": "text", "size": 30, "html": true, "sortable": false, "filterable": false },
-  "like" : { "displayName": $likesDisplayName, "type" : "number", "sortable": false, "filterable": false }
-})
-
-#set ($currentUser = $services.model.serialize($doc.documentReference, "default"))
-#set ($pageToCallReference = $services.model.resolveDocument("XWiki.Like.UserProfileUIX"))
-#set ($pageToCall = $xwiki.getDocument($pageToCallReference))
-#set ($queryParams = {
-  "user": $currentUser,
-  "livetable": "true",
-  "xpage": "plain",
-  "outputSyntax": "plain"
-})
-#set($options = {
-  'url': $pageToCall.getURL('get', $escapetool.url($queryParams))
+#set ($currentUser = $services.model.serialize($doc.documentReference, 'default'))
+#set ($liveDataConfig = {
+  'meta': {
+    'propertyDescriptors': [
+       {
+         'id': 'doc.title',
+         'displayer': 'link',
+         'sortable': false,
+         'filterable': false,
+         'name': $docTitleDisplayName
+       },
+       {
+         'id': 'doc.location',
+         'displayer': 'html',
+         'sortable': false,
+         'filterable': false,
+         'name': $docLocationDisplayName
+       },
+       {
+         'id': 'like',
+         'displayer': 'number',
+         'sortable': false,
+         'filterable': false,
+         'name': $likesDisplayName
+       }
+    ]
+  }
 })
-#livetable("likedPages" $columns $columnsProperties $options)
+##
+#set ($sourceParameters = $escapetool.url({
+  'resultPage': 'XWiki.Like.UserProfileUIX',
+  'user': $currentUser,
+  'livetable': 'true',
+  'xpage': 'plain',
+  'outputSyntax': 'plain'
+}))
+##
+{{liveData
+  id='likedPages'
+  source='liveTable'
+  sourceParameters="$sourceParameters"
+  properties='doc.title,doc.location,like'
+}}$jsontool.serialize($liveDataConfig){{/liveData}}
```

Comment:
Interestingly it doesn't seem that using LD is simpler than using the previous LT. There's actually more lines to write it seems... Isn't there a simpler way?

Also I see `source='liveTable'`. Does it mean we cannot yet use a LD source or are there other considerations (maybe some backward compat) that made you reuse the LT source?

Thanks

Reply:
> Interestingly it doesn't seem that using LD is simpler than using the previous LT

it seems actually a lot more complex seen that we also need java code...

Reply:
@vmassol as mentioned on #xwiki, we're reusing the live table source for now since we haven't rewritten it as a generic live data source (there's a bridge). As for the live data configuration, most of it should be automatically generated (when using an xclass) so we should be writing less lines of code than before, if not then there's probably a mistake.